appservice/src/main/java/nu/yona/server/admin/rest/AdminController.java

```diff
@@ -4,29 +4,52 @@
  *******************************************************************************/
 package nu.yona.server.admin.rest;
 
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+import javax.servlet.http.HttpServletRequest;
+
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.mvc.ControllerLinkBuilder;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.ResponseStatus;
 
+import nu.yona.server.DOSProtectionService;
+import nu.yona.server.properties.YonaProperties;
 import nu.yona.server.subscriptions.service.UserService;
 
 @Controller
 @RequestMapping(value = "/admin", produces = { MediaType.APPLICATION_JSON_VALUE })
 public class AdminController
 {
+	@Autowired
+	private DOSProtectionService dosProtectionService;
+
+	@Autowired
+	private YonaProperties yonaProperties;
 
 	@Autowired
 	private UserService userService;
 
 	@RequestMapping(value = "/requestUserOverwrite/", method = RequestMethod.POST)
-	@ResponseStatus(HttpStatus.OK)
-	public void setOverwriteUserConfirmationCode(@RequestParam String mobileNumber)
+	public ResponseEntity<String> setOverwriteUserConfirmationCode(@RequestParam String mobileNumber, HttpServletRequest request)
```

Comment:
See [my comment on the ticket.](http://jira.yona.nu/browse/YD-265). I don't think we need the ``dosProtectionService`` here.